src/showprogress.m

```diff
@@ -0,0 +1,25 @@
+% showprogress shows waitbars
+% Inputs
+% x, whichbar, varagin: see waitbar header for explanation
+% Feb 2017, Lemmer El Assal
+
+function fout = showprogress(x, whichbar, varargin)
+    global WAITBAR_TYPE;
+    fout = [];
+    if ~isempty(WAITBAR_TYPE)
+        switch WAITBAR_TYPE
+            case 1 % graphic waitbar
+                if nargin > 2
+                    fout = waitbar(x, whichbar, varargin);
+                else
+                    fout = waitbar(x, whichbar);
+                end
+            case 2 % text
+                if x > 0
+                    textprogressbar(x*100);
```

Comment:
Where is the `textprogressbar` function defined?
Add the `textprogressbar.m` file to the `external` folder (as this is not code that we wrote).